connectors/concur/src/main/java/com/vmware/connectors/concur/ConcurController.java

```diff
@@ -48,17 +54,35 @@
 
     private static final Logger logger = LoggerFactory.getLogger(ConcurController.class);
 
-
     private final WebClient rest;
     private final CardTextAccessor cardTextAccessor;
     private final Resource concurrRequestTemplate;
 
+    private final String clientId;
+    private final String clientSecret;
+    private final String oauthTokenUrl;
+
+    // Client credentials token fields.
+    private static final String CLIENT_ID = "client_id";
+    private static final String CLIENT_SECRET = "client_secret";
+    private static final String USERNAME = "username";
+    private static final String PASSWORD = "password";
+    private static final String CRED_TYPE = "credtype";
+    private static final String GRANT_TYPE = "grant_type";
+    private static final String BEARER = "Bearer ";
+
     @Autowired
     public ConcurController(WebClient rest,
                             CardTextAccessor cardTextAccessor,
+                            @Value("${concur.client-id}") final String clientId,
+                            @Value("${concur.client-secret}") final String clientSecret,
+                            @Value("${concur.oauth-instance-url}") final String oauthTokenUrl,
```

Comment:
Since this is now the baseUrl, take base url out of the metadata: https://github.com/vmware/connectors-workspace-one/blob/01232ebb30213aae79fdbc9d0d12079c0ed6422e/connectors/concur/src/main/resources/static/discovery/metadata.json#L3

And probably name this field (and setting) baseUrl and get rid of the baseUrl parameter in method parameters.

Reply:
Also, update the documentation (put something about setting this in the README -- for the docker container, see the airwatch readme for an example https://github.com/vmware/connectors-workspace-one/tree/01232ebb30213aae79fdbc9d0d12079c0ed6422e/connectors/airwatch#docker )

Reply:
(discussed in person) We can't take out the baseUrl b/c the host for the rest api is different than the host for the oauth api.